### PR TITLE
feat(fs extra): update APi to match `fs-extra 5.0.0`

### PR DESCRIPTION
**Pull request by alan-agius4**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: <<https://github.com/jprichardson/node-fs-extra/blob/master/CHANGELOG.md>>
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by shiftkey** [comment]
Seems fine 👍

File: types/fs-extra/index.d.ts [hunk 1]
```diff
@@ -254,7 +254,7 @@ export interface PathEntryStream {
     read(): PathEntry | null;
 }
 
-export type CopyFilter = ((src: string, dest: string) => boolean) | RegExp;
+export type CopyFilter = (src: string, dest: string) => boolean;
```
Comment (minestarks) [inline types/fs-extra/index.d.ts]: Is this a bugfix, or a change from v4?

Comment (alan-agius4) [reply types/fs-extra/index.d.ts]: Its a change from v4

https://github.com/jprichardson/node-fs-extra/blob/master/CHANGELOG.md

Comment (minestarks) [reply types/fs-extra/index.d.ts]: Thanks. Could you back up the old version to a new folder as described in the readme? https://github.com/DefinitelyTyped/DefinitelyTyped#i-want-to-update-a-package-to-a-new-major-version

**Review by minestarks — CHANGES_REQUESTED** [review]

**Comment by alan-agius4** [comment]
PR comments addressed

**Review by minestarks — APPROVED** [review]

**Pull request merged**